feat: support a port range on the `add` command

jacderida opened the pull request:
Previously we would not allow a custom port to be specified if more than one service was being
added. However, without autonat, people still need to open ports manually on their router, so the
node services they add will need to match this port range. Without the ability to specify a range,
they'd need to add the services one by one.

## Description

reviewpad:summary

grumbach commented on sn_node/src/node/messaging/anti_entropy.rs:
```diff
@@ -179,211 +186,78 @@ impl Node {
-            .await?;
+        // Check if we need to resend any messsages and who should we send it to.
+        let (bounced_msg, response_peer) = match kind {
+            AntiEntropyKind::Update { .. } => return Ok(cmds), // Nope, bail early
```
Do we need the field in `AntiEntropyKind::Update`?

joshuef replied:
`bounced_msg` you mean @grumbach ? No, the Update is purely about network knowledge. It's sent proactively or in response to a probe

grumbach reviewed (approved):
LGTM!

jacderida committed:
feat: support a port range on the `add` command

Previously we would not allow a custom port to be specified if more than one service was being
added. However, without autonat, people still need to open ports manually on their router, so the
node services they add will need to match this port range. Without the ability to specify a range,
they'd need to add the services one by one.
